packages/viewer/src/store/modules/app/index.ts

```diff
@@ -51,15 +51,26 @@ const initiateUrlParsing: AppState = {
 
 const parseLegacyUrlParams: AppState = {
     name: AppStateNames.LegacyParsing,
-    isFulfilled: () => !isLegacyParams(window?.location?.search),
+    isFulfilled: () => {
+        // Since it's not a ref here, it only checks it once and never check if it has changed
+        // we should have a small ref which contains windows.location
```

Comment:
I don't see what it would bring here to have a ref on `windows.location`, if there are some legacy params they are there from the start and won't "change" during the initialization.

Can you remove this comment, and if you really think something should change here create a ticket on our JIRA instead?